File: app/src/main/java/com/qweex/smashuphelper/Faction.java
package com.qweex.smashuphelper;


import android.os.Parcel;
import android.os.Parcelable;

public class Faction extends Persistable {
    Faction(String key) {
        super(key);
    }

    Faction(Parcel src) {
        super(src);
    }

    public static Parcelable.Creator<Faction> CREATOR = new Parcelable.Creator<Faction>() {
        @Override
        public Faction createFromParcel(Parcel in) {
            return new Faction(in);
        }

        @Override
        public Faction[] newArray(int size) {
            return new Faction[size];
        }
    };
}

File: smashuphelper/src/main/java/com/qweex/smashuphelper/Faction.kt
package com.qweex.smashuphelper


import android.os.Parcel
import android.os.Parcelable

class Faction : Persistable {
    internal constructor(key: String) : super(key) {}

    internal constructor(src: Parcel) : super(src) {}

    companion object CREATOR : Parcelable.Creator<Faction> {
        override fun createFromParcel(parcel: Parcel): Faction {
            return Faction(parcel)
        }

        override fun newArray(size: Int): Array<Faction?> {
            return arrayOfNulls(size)
        }
    }
}
